klient/constructor/deployment/constructor.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package deployment provides constructor types for appV1.Deployment
+package deployment
+
+import (
+	appsV1 "k8s.io/api/apps/v1"
+	"k8s.io/api/core/v1"
+	"sigs.k8s.io/e2e-framework/klient/constructor/meta"
+	"sigs.k8s.io/e2e-framework/klient/constructor/pod"
+)
+
+type Constructor struct {
+	deployment appsV1.Deployment
+}
+
+// Deployment is the initializer function for type deployment.Constructor
+func Deployment(
```

Comment:
I'm trying to think out loud about how we can scale this model without the need to write a 1:1. Is it possible to have a general constructor that takes kind/type? that will cover Deployment, StatefulSet, and DaemonSet. wdyt?